### PR TITLE
check collateral output feature using build-raw

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

**Review by mkoura — CHANGES_REQUESTED** [review]

File: cardano_node_tests/tests/test_plutus_v2_spend_raw.py [hunk 1]
```diff
@@ -1484,3 +1501,476 @@ def test_v1_attached_v2_reference(
...
+
+        # check that the right amount of collateral was spent and that the tokens were returned
+        assert (
+            cluster.get_address_balance(dst_addr.address, coin=tokens_rec[0].coin) == token_amount
+        ), f"Incorrect token balance for source address `{dst_addr.address}`"
+
+        assert (
+            cluster.get_address_balance(payment_addrs[1].address)
+            == dst_init_balance - total_collateral_amount
+        ), f"Incorrect Lovelace balance for source address `{dst_addr.address}`"
```
[the remark this answers is not in the document]
Comment (saratomaz) [reply cardano_node_tests/tests/test_plutus_v2_spend_raw.py:1753]:
```
# check that the right amount of collateral was spent and that the tokens were returned
lovelace_return_collateral = tx_output_redeem.return_collateral_txouts[0]
token_return_collateral = tx_output_redeem.return_collateral_txouts[1]
assert (
    token_return_collateral.amount == token_amount and
    token_return_collateral.coin == tokens_rec[0].coin
), f"Incorrect token balance for source address `{dst_addr.address}`"

assert (
    lovelace_return_collateral.amount == dst_init_balance - total_collateral_amount
), f"Incorrect Lovelace balance for source address `{dst_addr.address}`"
```

File: cardano_node_tests/tests/test_plutus_v2_spend_raw.py [hunk 2]
```diff
@@ -1484,3 +1501,476 @@ def test_v1_attached_v2_reference(
...
+
+
+@pytest.mark.testnets
+class TestNegativeCollateralOutput:
```
[the remark this answers is not in the document]
Comment (saratomaz) [reply cardano_node_tests/tests/test_plutus_v2_spend_raw.py]: moved to pluts_v2_mint, also did a refactor and joined the new two test about negative collaterals, because the error is the same :)

File: cardano_node_tests/tests/test_plutus_v2_spend_raw.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkoura) [reply cardano_node_tests/tests/test_plutus_v2_spend_raw.py:1756]: That doesn't query the UTxO, it just compares data in data structures created in the test.
I was thinking about something like
```diff
diff --git a/cardano_node_tests/tests/test_plutus_v2_spend_raw.py b/cardano_node_tests/tests/test_plutus_v2_spend_raw.py
index 7873cf19..b81e6de8 100644
--- a/cardano_node_tests/tests/test_plutus_v2_spend_raw.py
+++ b/cardano_node_tests/tests/test_plutus_v2_spend_raw.py
@@ -1657,8 +1657,8 @@ class TestCollateralOutput:
         amount = 2_000_000
         token_amount = 100
 
-        total_collateral_amount = redeem_cost.collateral
         amount_for_collateral = redeem_cost.collateral * 4
+        return_collateral_amount = amount_for_collateral - redeem_cost.collateral
 
         # creat the token
         token_rand = clusterlib.get_rand_str(5)
@@ -1685,8 +1685,6 @@ class TestCollateralOutput:
             tokens_collateral=tokens_rec,
         )
 
-        dst_init_balance = cluster.get_address_balance(dst_addr.address)
-
         # spend the locked UTxO
 
         plutus_txins = [
@@ -1713,7 +1711,7 @@ class TestCollateralOutput:
         txouts_return_collateral = [
             clusterlib.TxOut(
                 address=dst_addr.address,
-                amount=amount_for_collateral - total_collateral_amount,
+                amount=return_collateral_amount,
             ),
             clusterlib.TxOut(
                 address=dst_addr.address, amount=token_amount, coin=tokens_rec[0].coin
@@ -1728,7 +1726,7 @@ class TestCollateralOutput:
             script_txins=plutus_txins,
             script_valid=False,
             return_collateral_txouts=txouts_return_collateral,
-            total_collateral_amount=total_collateral_amount,
+            total_collateral_amount=redeem_cost.collateral,
             join_txouts=False,
         )
         tx_signed_redeem = cluster.sign_tx(
@@ -1743,14 +1741,18 @@ class TestCollateralOutput:
         )
 
         # check that the right amount of collateral was spent and that the tokens were returned
-        assert tx_output_redeem.return_collateral_txouts
-        lovelace_return_collateral = tx_output_redeem.return_collateral_txouts[0]
-        token_return_collateral = tx_output_redeem.return_collateral_txouts[1]
+        txid_redeem = cluster.get_txid(tx_body_file=tx_output_redeem.out_file)
+        return_col_utxos = cluster.get_utxo(txin=f"{txid_redeem}#{len(txouts_redeem)}")
+        assert return_col_utxos, "Return collateral UTxO was not created"
+
+        return_col_utxos_lovelace = [
+            u for u in return_col_utxos if u.coin == clusterlib.DEFAULT_COIN
+        ]
         assert (
-            token_return_collateral.amount == token_amount
-            and token_return_collateral.coin == tokens_rec[0].coin
-        ), f"Incorrect token balance for source address `{dst_addr.address}`"
+            return_col_utxos_lovelace[0].amount == return_collateral_amount
+        ), f"Incorrect balance for collateral return address `{dst_addr.address}`"
 
+        return_col_utxos_token = [u for u in return_col_utxos if u.coin == tokens_rec[0].coin]
         assert (
-            lovelace_return_collateral.amount == dst_init_balance - total_collateral_amount
-        ), f"Incorrect Lovelace balance for source address `{dst_addr.address}`"
+            return_col_utxos_token[0].amount == tokens_rec[0].amount
+        ), f"Incorrect token balance for collateral return address `{dst_addr.address}`"
```

[the remark this answers is not in the document]
Comment (mkoura) [reply cardano_node_tests/tests/test_plutus_v2_spend_raw.py:1756]: The `cluster.get_utxo(txin=f"{txid_redeem}#{len(txouts_redeem)}")` also checks that the following is true - https://github.com/input-output-hk/cardano-ledger/issues/2885#issuecomment-1168627236

**Review by mkoura — APPROVED** [review]
